src/compiler/program.ts

```diff
@@ -107,24 +107,51 @@ namespace ts {
             if (directoryPath.length > getRootLength(directoryPath) && !directoryExists(directoryPath)) {
                 const parentDirectory = getDirectoryPath(directoryPath);
                 ensureDirectoriesExist(parentDirectory);
-                if (compilerHost.createDirectory) {
-                    compilerHost.createDirectory(directoryPath);
+                (compilerHost.createDirectory || system.createDirectory)(directoryPath);
+            }
+        }
+
+        function writeFile(fileName: string, data: string, writeByteOrderMark: boolean, onError?: (message: string) => void) {
+            try {
+                performance.mark("beforeIOWrite");
+
+                // PERF: Checking for directory existence is expensive.
+                // Instead, assume the directory exists and fall back
+                // to creating it if the file write fails.
+                // NOTE: If patchWriteFileEnsuringDirectory has been called,
+                // the file write will do its own directory creation and
+                // the ensureDirectoriesExist call will always be redundant.
+                try {
+                    writeFileWorker(fileName, data, writeByteOrderMark);
                 }
-                else {
-                    system.createDirectory(directoryPath);
+                catch (_) {

```

Comment:
Nit: `catch` clauses can omit their variable.

Reply:
I thought I was getting squiggled for that, but I'll try again.  Do I omit the parens too?

Reply:
Yeah, `catch {`